### DISCUSSION TITLE
Question about password to be passed as a URL parameter to unlock albums

### DISCUSSION THREAD

**Discussion by qwrty-ftw** [LycheeOrg/Lychee#2339] [Q&A] [answered]
Hello, 
I'm running last version with docker, but I cannot figured out how this option works ?
I have activated the option (unlock_password_photos_with_url_param = 1) , but how can I generate my link with the password in the URL for my album ?
Thank you !

Lychee 5.1.2

**Comment by ildyria** [comment]
> how can I generate my link with the password in the URL for my album ?

In theory the link is provided in the share button of the album (if I am not wrong).

**Comment by AndyYangUK** [comment]
DId you ever find a solution for this @qwrty-ftw?

**Reply by ildyria** [reply]
I think this is currently not supported in v6. I will see to fix it.

**Comment by ildyria** [comment, marked as answer]
So further research shows me that this is not possible in the current state of Lychee. Password of albums are stored bcrypt-hashed.
As a result, it is not possible to generate a URL with the password already pre-filled. What could be possible though is to provide the ?password= suffix to hint that something needs to be filled by the user before giving the link away.

I will need to check later whether this query parameter is used when accessing an album.

**Reply by ildyria** [reply]
Now available on master: https://github.com/LycheeOrg/Lychee/commit/1d7fbf796cf1b3b5563b82912e42f99ff9ed228d
Note that you need to add the `?password=.....` yourself to the url.
